src/query/shorthand.ts

```diff
@@ -49,6 +50,59 @@ export function vlSpec(vlspec: ExtendedUnitSpec,
   return spec(specQ);
 }
 
+const CHANNEL_SUPPORTS_AXIS: Dict<boolean> =
+  [Channel.X, Channel.Y, Channel.ROW, Channel.COLUMN]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+const CHANNEL_SUPPORTS_LEGEND: Dict<boolean> =
+  [Channel.COLOR, Channel.OPACITY, Channel.SIZE, Channel.SHAPE]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+const CHANNEL_SUPPORTS_SCALE: Dict<boolean> =
+  [Channel.X, Channel.Y, Channel.COLOR, Channel.OPACITY, Channel.OPACITY, Channel.ROW, Channel.COLUMN, Channel.SIZE, Channel.SHAPE]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+const CHANNEL_SUPPORTS_SORT: Dict<boolean> =
+  [Channel.X, Channel.Y, Channel.PATH, Channel.ORDER]
+    .reduce((m, channel) => {
+      m[channel] = true;
+      return m;
+    }, {} as Dict<boolean>);
+
+export const CHANNEL_SUPPORTS_PROPERTY =
```

Comment:
I'm sure there are smart way to make this chunk of code shorter.  Either 

`PROPERTY_SUPPORTED_CHANNELS[prop][channel] `
or
`CHANNEL_SUPPORTED_PROPERTY[channel][prop]`

would work.  You just need to think more.